File: 060_Permutation_Sequence/060_1.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.Deque;
import java.util.Queue;
import java.util.ArrayDeque;
import java.util.List;
import java.util.ArrayList;
import java.util.LinkedList;

class Solution {
    public String getPermutation(int n, int k) {
        k--;
        int[] factorial = new int[n];
        factorial[0] = 1;
        for (int i = 1; i < n; i++) {
            factorial[i] = factorial[i - 1] * i;
        }

        List<Integer> nums = new LinkedList<>();
        for (int i = 1; i <= n; i++) {
            nums.add(i);
        }

        StringBuilder sb = new StringBuilder();
        for (int i = n - 1; i >= 0; i--) {
            int index = k / factorial[i];
            sb.append(nums.remove(index));
            k -= index * factorial[i];
        }
        return sb.toString();
    }
}

public class leetcode {
    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.getPermutation(3, 3));
    }
}

File: 060_Permutation_Sequence/060_1.py
from typing import List, Dict


class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        k -= 1
        factorial = [1]
        for i in range(1, n):
            factorial.append(factorial[-1] * i)
        
        nums = []
        for i in range(1, n+1):
            nums.append(i)
        
        res = []
        for i in range(n - 1, -1, -1):
            index = k // factorial[i]
            res.append(str(nums[index]))
            nums.pop(index)
            k -= index * factorial[i]
        return "".join(res)
        

if __name__ == "__main__":
    s = Solution()
    print(s.getPermutation(3, 3))
